lib/model/rwfolder.go

```diff
@@ -1505,7 +1519,7 @@ func (p *rwFolder) moveForConflict(name string) error {
 		err = nil
 	}
 	if p.maxConflicts > -1 {
-		matches, gerr := osutil.Glob(withoutExt + ".sync-conflict-????????-??????" + ext)
+		matches, gerr := osutil.Glob(withoutExt + "????????-??????" + ext)
```

Comment:
This cleans up old conflict files when we've reached the maximum allowed number. But `withoutExt` here contains the device ID of the "winner", so you're only matching old conflict files from that same device. The cleanup code should probably remove all extraneous conflict files regardless of device.